src/replica/models/replicaManager.ts

```diff
@@ -124,23 +130,21 @@ export class ReplicaManager {
         layerId,
         geometryType,
         timestamp,
-        url: urls,
+        urls,
       };
     });
   }
 
-  private getUrlHeader(): string {
-    const { protocol, host, port } = this.objectStorageConfig;
-    return `${protocol}://${host}:${port}`;
-  }
-
   private getReplicaUrls(replicaWithFiles: ReplicaWithFiles): string[] {
     const { bucketName, layerId, geometryType, files } = replicaWithFiles;
     const { projectId } = this.objectStorageConfig;
     let bucketOrProjectIdWithBucket = bucketName;
     if (!isStringUndefinedOrEmpty(projectId)) {
       bucketOrProjectIdWithBucket = `${projectId}:${bucketName}`;
     }
-    return files.map((file) => `${this.urlHeader}/${bucketOrProjectIdWithBucket}/${layerId}/${geometryType}/${file.fileId}`);
+    return files.map((file) => {
+      const filePath = join(bucketOrProjectIdWithBucket, layerId.toString(), geometryType, file.fileId);
+      return `${this.urlHeader}/${filePath}`;
```

Comment:
can we use join for urlHeader too ?
`return join(this.urlHeader, bucketOrProjectIdWithBucket, layerId.toString(), geometryType, file.fileId);`?

Reply:
nope, because it's not a path. for instance the `//` will become a single `/`

Reply:
have you searched for a maintained package for building URLs?